### DISCUSSION TITLE
getServerSideProps is ignored

### DISCUSSION THREAD

**Discussion by amsterdamharu** [vercel/next.js#11009] [Help] [answered]
Given the following code it looks like getServerSideProps is ignored completely.

```javascript
import React from "react";
import Layout from "../../components/Layout";

function Post(props) {
  console.log("in render");
  return (
    <Layout title={"post title"}>
      <pre>{JSON.stringify(props, undefined, 2)}</pre>
      {/* this shows
          {
            "url": {
              "query": {
                "id": "8"
              },
              "pathname": "/post/[id]",
              "asPath": "/post/8"
            }
          }      
      */}
    </Layout>
  );
}

export async function getServerSideProps(needQuery) {
  console.log("never see this", needQuery);
  return { props: { dosNot: "matter, never called" } };
}

export default Post;
```

On the server it logs "in render" once and on the client it logs "in render" twice and one warning about accessing the url property once on server and client.

I am using next version [9.3.0](https://github.com/harm-meijer/nextjs-example/blob/620d30aed947cbc59c2453efe5f38c1ca21773d3/yarn.lock#L3760-L3760)

Full code is [here](https://github.com/harm-meijer/nextjs-example/tree/refactor) (refactor branch) and I start it with `yarn dev`.

Just noticed that not only does it not work, it also breaks single page application and causes a page reload when navigating from list of posts to single post, that didn't happen in master branch.

**Comment by ijjk** [comment, marked as answer]
Hi, it looks like you aren't exporting `getServerSideProps` from your [pages/post/[id].js](https://github.com/harm-meijer/nextjs-example/blob/620d30aed947cbc59c2453efe5f38c1ca21773d3/pages/post/%5Bid%5D.js#L2) file. This should be `export { default, getServerSideProps } from './index'`

`getServerSideProps` is called correctly on the above repo for the `/post` page itself since you are exporting it from that page as required.

A page being reloaded from a list of posts would be caused from invalid usage of `next/link` or `next/router`. The `href` value must always be the `page` on the filesystem e.g. `/post/[id]` and `as` is the URL `/post/post-1`. See [our docs](https://nextjs.org/docs/api-reference/next/link#dynamic-routes) for more info

**Reply by EhanDuan** [reply]
Hi, I have a similar issue that `console.log` does not work in the asyn`getServerSideProps` function, other codes working.

I checked your reply. We need also `export { default, getServerSideProps } from './index`. But I check the docs for `getServerSideProps`, it does not mention about `export { getServerSideProps}`.

So my question is how do we know we need to `export { getServerSideProps}`. What does it depend on?

Thanks for your help.
